qemu/tests/s390x_cpu_topology_test.py

```diff
@@ -0,0 +1,41 @@
+from provider.cpu_utils import check_if_vm_vcpu_topology_match
+
+
+def run(test, params, env):
+    """
+    Check guest gets correct vcpu num, cpu cores, processors, sockets, siblings
+
+    1) Boot guest with options: -smp n,cores=x,threads=y,sockets=z...
+    2) Check cpu topology is expected as set up
+
+    :param test: QEMU test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+
+    def check(p_name, exp, check_cmd):
+        """
+        Check the cpu property inside guest
+        :param p_name: Property name
+        :param exp: The expect value
+        :param check_cmd: The param to get check command
+        """
+        res = session.cmd_output(check_cmd).strip()
+        if int(res) != int(exp):
+            test.fail('The vcpu %s number inside guest is %s,'
+                      ' while it is set to %s' % (p_name, res, exp))
+
+    vm_name = params['main_vm']
+    os_type = params['os_type']
+    vm = env.get_vm(vm_name)
+    session = vm.wait_for_login()
+    if not check_if_vm_vcpu_topology_match(session, os_type, vm.cpuinfo,
+                                           test, vm.devices):
+        test.fail('CPU topology of guest is incorrect.')
+    vcpu_sockets = params['vcpu_sockets']
+    if params.get('check_core_id_cmd'):
+        for cpu_id in list(range(params['smp'])):
```

Comment:
When I define `check_core_id_cmd = 'cat /sys/devices/system/cpu/cpu%s/topology/core_id'` for this variant I run into another issue:
```
virttest.utils_params.ParamNotFound: Mandatory parameter 'smp' is missing. Check your cfg files for typos/mistakes
```